lib/mllp/packet_framer.ex

```diff
@@ -74,6 +76,66 @@ defmodule MLLP.PacketFramer do
               end
             end
 
+            def handle_packet(
+                  unquote(carriage_return),
+                  %FramingContext{current_message_type: unquote(message_type)} = state
+                ) do
+
+              message_type_value = unquote(message_type)
+              check = byte_size(state.receiver_buffer) - 1
+
+              case state.receiver_buffer do
+                <<message::binary-size(check), unquote(fs_sep)>> ->
+                  message_type_atom = get_message_type(message_type_value, message)
+
+                  {:ok, new_state} =
+                    state.dispatcher_module.dispatch(
+                      message_type_atom,
+                      message,
+                      %{
+                        state
+                        | # save leftovers to prepend to next packet
+                          receiver_buffer: "",
+                          current_message_type: nil
+                      }
+                    )
+
+                  {:ok, new_state}
+
+                _ ->
+                  {:ok,
+                   %{
+                     state
+                     | receiver_buffer: state.receiver_buffer <> unquote(carriage_return),
+                       current_message_type: message_type_value
+                   }}
+              end
+            end
+
+            def handle_packet(
```

Comment:
wouldn't this case be handled by handle_packet(packet,...) function at line 139

Reply:
You are right, beautiful catch ❤️